pkg/core/mempool/mem_pool.go

```diff
@@ -609,3 +610,34 @@ func (mp *Pool) removeConflictsOf(tx *transaction.Transaction) {
 		}
 	}
 }
+
+// TryFindTransaction tries to find a transaction in the memory pool.
+// It searches through all the verified main and fallback transactions.
+func (mp *Pool) TryFindTransaction(hash util.Uint256) (*transaction.Transaction, bool) {
+	mp.lock.RLock()
+	defer mp.lock.RUnlock()
+
+	for i := range mp.verifiedTxes {
+		data := mp.verifiedTxes[i].data.(*payload.P2PNotaryRequest)
+		if data.MainTransaction.Hash() == hash {
+			return data.MainTransaction, true
+		} else if data.FallbackTransaction.Hash() == hash {
+			return data.FallbackTransaction, true
+		}
+	}
+	return nil, false
+}
+
+// GetVerifiedPoolTx returns a map containing all the verified transactions.
+// The main transactions are the keys, and the corresponding values are slices of fallback transactions.
+func (mp *Pool) GetVerifiedPoolTx() map[util.Uint256][]util.Uint256 {
+	mp.lock.RLock()
+	defer mp.lock.RUnlock()
+
+	r := make(map[util.Uint256][]util.Uint256)
+	for i := range mp.verifiedTxes {
+		data := mp.verifiedTxes[i].data.(*payload.P2PNotaryRequest)
+		r[data.MainTransaction.Hash()] = append(r[data.MainTransaction.Hash()], data.FallbackTransaction.Hash())
+	}
+	return r
+}
```

Comment:
And instead of this funtion we can have something like `(*Pool).IterateVerifiedTransactions(f func(tx *transaction.Transaction, data any))`. Of course, this function needs to be invoked with care because it holds read lock over the whole mempool, but we can specify this in the documentation.

@roman-khimov, what do you think? Or maybe I'm too worried about mempool isolation and we can just keep these functions as is.

Reply:
@AnnaShaleva what if to add function like:
```
func (mp *Pool) GetVerifiedPoolData() map[any][]any {
	mp.lock.RLock()
	defer mp.lock.RUnlock()

	r := make(map[any][]any)
	for i := range mp.verifiedTxes {
		if mp.verifiedTxes[i].data != nil {
			data := mp.verifiedTxes[i].data
			r[data] = append(r[data], data)
		}
	}
	return r
}
```
And than in rpc server func getRawNotaryPool extract only tx hashes?
For me it looks more clear and safe. 
What do you think?

Reply:
It was an invalid copy-paste. I meant to return slice of mp.verifiedTxes[i].data.
Anyway, I've done it as you suggested.